pkg/cosign/remote.go

```diff
@@ -45,20 +49,53 @@ func Descriptors(ref name.Reference) ([]v1.Descriptor, error) {
 	return m.Layers, nil
 }
 
-func Upload(signature, payload []byte, dstTag name.Reference, cert, chain string) error {
-	l := &staticLayer{
-		b:  payload,
-		mt: "application/vnd.dev.cosign.simplesigning.v1+json",
-	}
-	base, err := remote.Image(dstTag, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+// SignatureImage
+func SignatureImage(dstTag name.Reference, opts ...remote.Option) (v1.Image, error) {
+	base, err := remote.Image(dstTag, opts...)
 	if err != nil {
 		if te, ok := err.(*transport.Error); ok {
 			if te.StatusCode != http.StatusNotFound {
-				return te
+				return nil, te
 			}
 			base = empty.Image
 		} else {
-			return err
+			return nil, err
+		}
+	}
+	return base, nil
+}
+
+func Upload(ctx context.Context, signature, payload []byte, dst name.Reference, cert, chain string, dupeDetector signature.Verifier) (uploadedSig []byte, err error) {
+	l := &staticLayer{
+		b:  payload,
+		mt: SimpleSigningMediaType,
+	}
+
+	base, err := SignatureImage(dst, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	if err != nil {
+		return nil, err
+	}
+
+	if dupeDetector != nil {
+		signedDigest, err := l.Digest()
```

Comment:
style: maybe move this out to a helper?

if dupeDetector != nil {
  detectDupes(dupedetector, base)
}

Reply:
done